### PR TITLE
Non json resteasy reactive

### PR DESCRIPTION
**Pull request by jcarranzan**
### Summary

Create test cases that evolve non-JSON payload in REST endpoints as described in the Jira task:
https://issues.redhat.com/browse/QQE-343

1. XML scenario:
Scenario: Verify that the endpoint properly consumes and produces XML payloads.

- Check the endpoint that returns a valid XML.
- Send a request with an XML payload to the endpoint.
- Validate the response.
- Ensure the response's content matches the expected from the XML structure.

2. YAML scenario

- Check the endpoint that returns a Yaml from a Yaml file.
- Send a request with a Yaml payload from a file to an endpoint.
- Validate the response 
- Ensure the response's content matches the expected from the YAML content.

Please select the relevant options.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] Dependency update
- [ ] Refactoring
- [ ] Backport
- [x] New scenario (non-breaking change which adds functionality)
- [ ] This change requires a documentation update
- [ ] This change requires execution against OCP (use `run tests` phrase in comment)

### Checklist:
- [x] Methods and classes used in PR scenarios are meaningful
- [x] Commits are well encapsulated and follow [the best practices](https://cbea.ms/git-commit/)

### CODE REVIEW

File: http/http-advanced-reactive/src/main/java/io/quarkus/ts/http/advanced/reactive/CityResource.java
Comment (michalvavrik) [inline http/http-advanced-reactive/src/main/java/io/quarkus/ts/http/advanced/reactive/CityResource.java:83]: Why is `handleYamlPostRequest` producing application/json?

Comment (jcarranzan) [reply http/http-advanced-reactive/src/main/java/io/quarkus/ts/http/advanced/reactive/CityResource.java:83]: Why not? I mean the request already uses a YAML payload, and also in the @GET we already checked that it could be @Produce a Yaml here:    [ @Produces(APPLICATION_YAML)](https://github.com/quarkus-qe/quarkus-test-suite/blob/7db3158f3b330b551335d884f02c8bb0c6cc7f34/http/http-advanced-reactive/src/main/java/io/quarkus/ts/http/advanced/reactive/CityResource.java#L100)

Comment (michalvavrik) [reply http/http-advanced-reactive/src/main/java/io/quarkus/ts/http/advanced/reactive/CityResource.java:83]: ok

Comment (mjurc) [reply http/http-advanced-reactive/src/main/java/io/quarkus/ts/http/advanced/reactive/CityResource.java:83]: Tbh it's kinda confusing at first gaze - why not just remove `Yaml` from the path? If people come back to this test in the future, they'll be scratching their head too :]

**Review by mjurc — CHANGES_REQUESTED** [review]
Are you planning to add the image payload test too btw?

File: http/http-advanced-reactive/src/main/java/io/quarkus/ts/http/advanced/reactive/CityResource.java
Comment (michalvavrik) [reply http/http-advanced-reactive/src/main/java/io/quarkus/ts/http/advanced/reactive/CityResource.java:83]: truth be told, I didn't want to argue, but I think name should reflect that yaml is only consumed, not produced. like `handleConsumedYamlPostRequest` (just better :-)).

I also think it is rather rare to consume YAML and produce JSON and there is no scenario AFAICT that tests both consume/produce YAML, but I'll respect @jcarranzan preference.

File: http/http-advanced-reactive/src/test/java/io/quarkus/ts/http/advanced/reactive/NonJsonPayLoadIT.java
Comment (michalvavrik) [inline http/http-advanced-reactive/src/test/java/io/quarkus/ts/http/advanced/reactive/NonJsonPayLoadIT.java:72]: this is hard to read, please use string block for all these long strings

Comment (michalvavrik) [reply http/http-advanced-reactive/src/test/java/io/quarkus/ts/http/advanced/reactive/NonJsonPayLoadIT.java:72]: hmm, unless you need to backport it, but I don't think you do

File: http/http-advanced-reactive/src/test/java/io/quarkus/ts/http/advanced/reactive/NonJsonPayLoadIT.java
Comment (michalvavrik) [inline http/http-advanced-reactive/src/test/java/io/quarkus/ts/http/advanced/reactive/NonJsonPayLoadIT.java:164]: why not leave it to RestAssured? here https://www.james-willett.com/rest-assured-serialization-with-json-and-xml/#xml-serialization it looks like XML serialization should work.

File: http/http-advanced-reactive/src/main/java/io/quarkus/ts/http/advanced/reactive/CityListWrapperSerializer.java
Comment (michalvavrik) [inline http/http-advanced-reactive/src/main/java/io/quarkus/ts/http/advanced/reactive/CityListWrapperSerializer.java:13]: we use JBOSS logger, right?

File: http/http-advanced-reactive/src/main/java/io/quarkus/ts/http/advanced/reactive/CityResource.java
Comment (michalvavrik) [inline http/http-advanced-reactive/src/main/java/io/quarkus/ts/http/advanced/reactive/CityResource.java:85]: If this is `string`, there is no serialization in place, how does this actually test YAML content type consumption if you do serialization/deserialization yourself?

Please let me give you example: change `@Consumes` to JSON  or plain text and let me know if there is any difference. Thank you

File: http/http-advanced-reactive/src/test/java/io/quarkus/ts/http/advanced/reactive/NonJsonPayLoadIT.java
Comment (michalvavrik) [inline http/http-advanced-reactive/src/test/java/io/quarkus/ts/http/advanced/reactive/NonJsonPayLoadIT.java:131]: OS version :-) you prefer this over removing white spaces? you won't need to do that for response part if you avoid pretty string.

File: http/http-advanced-reactive/src/test/java/io/quarkus/ts/http/advanced/reactive/NonJsonPayLoadIT.java
Comment (michalvavrik) [inline http/http-advanced-reactive/src/test/java/io/quarkus/ts/http/advanced/reactive/NonJsonPayLoadIT.java:126]: I'd like to point out that you don't actually test YAML. You test that you can send String from endpoint, consume string and because you manually serialized DTO to YAML, it is YAML.  The condition is always true because you do it manually.

File: http/http-advanced-reactive/src/main/java/io/quarkus/ts/http/advanced/reactive/CityResource.java
Comment (jcarranzan) [reply http/http-advanced-reactive/src/main/java/io/quarkus/ts/http/advanced/reactive/CityResource.java:83]: Yes, looking at it again, you're right I will tweak it to change the path and the name to `handleConsumedYamlPostRequest`

**Comment by jcarranzan** [comment]
> Are you planning to add the image payload test too btw?

Yes, I will push it today if there are no setbacks.

File: http/http-advanced-reactive/src/main/java/io/quarkus/ts/http/advanced/reactive/CityListWrapperSerializer.java
Comment (jcarranzan) [reply http/http-advanced-reactive/src/main/java/io/quarkus/ts/http/advanced/reactive/CityListWrapperSerializer.java:13]: Oh, I see, in vert.x test suite we used to use slf4j, changing it to JBOSS...

File: http/http-advanced-reactive/src/test/java/io/quarkus/ts/http/advanced/reactive/NonJsonPayLoadIT.java
Comment (jcarranzan) [reply http/http-advanced-reactive/src/test/java/io/quarkus/ts/http/advanced/reactive/NonJsonPayLoadIT.java:72]: Yes, I'll use string block better.

File: http/http-advanced-reactive/src/test/java/io/quarkus/ts/http/advanced/reactive/NonJsonPayLoadIT.java
Comment (jcarranzan) [reply http/http-advanced-reactive/src/test/java/io/quarkus/ts/http/advanced/reactive/NonJsonPayLoadIT.java]: good example, it works.

File: http/http-advanced-reactive/src/test/java/io/quarkus/ts/http/advanced/reactive/NonJsonPayLoadIT.java
Comment (jcarranzan) [reply http/http-advanced-reactive/src/test/java/io/quarkus/ts/http/advanced/reactive/NonJsonPayLoadIT.java]: I removed it.

File: http/http-advanced-reactive/src/test/java/io/quarkus/ts/http/advanced/reactive/NonJsonPayLoadIT.java
Comment (michalvavrik) [reply http/http-advanced-reactive/src/test/java/io/quarkus/ts/http/advanced/reactive/NonJsonPayLoadIT.java:83]: `testExpectedYamlResponse` is acceptable now. I happen to disagree

File: http/http-advanced-reactive/src/test/java/io/quarkus/ts/http/advanced/reactive/NonJsonPayLoadIT.java
Comment (michalvavrik) [inline http/http-advanced-reactive/src/test/java/io/quarkus/ts/http/advanced/reactive/NonJsonPayLoadIT.java:126]: I'd prefer Howlin Wolf :/

No, now seriously, this image is not present in git project, therefore this test must always fail.

File: http/http-advanced-reactive/src/test/java/io/quarkus/ts/http/advanced/reactive/NonJsonPayLoadIT.java
Comment (michalvavrik) [inline http/http-advanced-reactive/src/test/java/io/quarkus/ts/http/advanced/reactive/NonJsonPayLoadIT.java:127]: I apologize for my gruelling review, but I'm worried this could be true even when byte array is empty? Anyway, only way to assure it works properly is to compare actual byte arrays.

Maybe I am missing something, please let me know if you disagree.

Comment (michalvavrik) [reply http/http-advanced-reactive/src/test/java/io/quarkus/ts/http/advanced/reactive/NonJsonPayLoadIT.java:127]: ha! exactly in a fashion you are doing it in the next test would be great!

File: http/http-advanced-reactive/src/main/java/io/quarkus/ts/http/advanced/reactive/CityResource.java
Comment (michalvavrik) [inline http/http-advanced-reactive/src/main/java/io/quarkus/ts/http/advanced/reactive/CityResource.java:64]:
```suggestion
    public String createCity(CityListDTO dto) {
```

File: http/http-advanced-reactive/src/main/java/io/quarkus/ts/http/advanced/reactive/CityResource.java
Comment (michalvavrik) [inline http/http-advanced-reactive/src/main/java/io/quarkus/ts/http/advanced/reactive/CityResource.java:97]: This looks like a smoke test for `YAMLMapper` as deserializing it as object is lenient. While I think it's nice to know that the `YAMLMapper` works in native, I don't think this is how users should write it. 

Microservice won't have one endpoint, right? It depends on use case, but more often than not will endpoints consume same content type (at least we can say more than one)? If your answer is yes, than having boilerplate in every endpoint method doesn't make sense to me. That is why I think users should consume DTOs unless in special cases. I think you test transfer of string and if you didn't deserialize it with YamlMapper, it could as well be JSON etc. and the test would pass with same consumed content type.

In short: there is not a single endpoint that consumes DTO and I disagree with it.

Comment (jcarranzan) [reply http/http-advanced-reactive/src/main/java/io/quarkus/ts/http/advanced/reactive/CityResource.java:88]: Well, in fact with my last changes, now there is an endpoint that consumes DTO https://github.com/quarkus-qe/quarkus-test-suite/blob/17584497b1ca70f5a5382d8c3dd363ab9f7fafcf/http/http-advanced-reactive/src/main/java/io/quarkus/ts/http/advanced/reactive/CityResource.java#L65

Comment (jcarranzan) [reply http/http-advanced-reactive/src/main/java/io/quarkus/ts/http/advanced/reactive/CityResource.java:88]: oh srry , this  @Consumes(MediaType.APPLICATION_XML)..., ok I will try to think it around to change it.

Comment (jcarranzan) [reply http/http-advanced-reactive/src/main/java/io/quarkus/ts/http/advanced/reactive/CityResource.java]: With the last changes I think this thread we could mark it as resolved, wdyt @michalvavrik ?

File: http/http-advanced-reactive/src/main/java/io/quarkus/ts/http/advanced/reactive/YamlProvider.java
Comment (michalvavrik) [inline http/http-advanced-reactive/src/main/java/io/quarkus/ts/http/advanced/reactive/YamlProvider.java:23]: interesting, so given you didn't specify `@Produces`  this will be applied to all content types? Are you sure this is never applied to `public CityListDTO getCities() {`?

**Review by michalvavrik — COMMENTED** [review]
LGTM if you can answer well my comment, thanks

File: http/http-advanced-reactive/src/main/java/io/quarkus/ts/http/advanced/reactive/YamlProvider.java
Comment (jcarranzan) [reply http/http-advanced-reactive/src/main/java/io/quarkus/ts/http/advanced/reactive/YamlProvider.java:23]: Well, it will be applied to all resource methods that consume "application/yaml".
If a method resource @produce another different content type , it takes precedence over the general provider.
In the case that you mentioned,it is:
```
@Produces(value = MediaType.APPLICATION_XML)
    public CityListDTO getCities() {
```

Comment (michalvavrik) [reply http/http-advanced-reactive/src/main/java/io/quarkus/ts/http/advanced/reactive/YamlProvider.java:23]: The `MessageBodyWriter` javadoc only speaks about `produces`, not `consumes` - https://jakarta.ee/specifications/platform/8/apidocs/javax/ws/rs/ext/messagebodywriter So I don't say you are not right, but I'm not sure if this is expected behavior.

**Review by michalvavrik — APPROVED** [review]

File: http/http-advanced-reactive/src/main/java/io/quarkus/ts/http/advanced/reactive/YamlProvider.java
Comment (jcarranzan) [reply http/http-advanced-reactive/src/main/java/io/quarkus/ts/http/advanced/reactive/YamlProvider.java:24]: added` @Produces("application/yaml")` to the provider to be more consistent. Thanks @michalvavrik .

**Review by mjurc — APPROVED** [review]

**Review by mjurc — CHANGES_REQUESTED** [review]
@jcarranzan thanks for the coverage, code LGTM, but please squash the commits and update the commit message to be more helpful

**Pull request merged**
